File: year2022/day22/direction.go
package year2022day22

type Direction int

const RigthDirection Direction = 0
const DownDirection Direction = 1
const LeftDirection Direction = 2
const UpDirection Direction = 3

func (direction Direction) String() string {
	if direction == UpDirection {
		return "Up"
	} else if direction == RigthDirection {
		return "Right"
	} else if direction == DownDirection {
		return "Down"
	} else if direction == LeftDirection {
		return "Left"
	}

	return "Unknown"
}

File: year2023/day18/src/direction.rs
#[derive(Debug, PartialEq)]
pub enum Direction {
    Up,
    Right,
    Down,
    Left,
}

impl Direction {
    pub fn from_char(c: char) -> Result<Direction, ()> {
        match c {
            'U' => Ok(Direction::Up),
            'R' => Ok(Direction::Right),
            'D' => Ok(Direction::Down),
            'L' => Ok(Direction::Left),
            _ => Err(()),
        }
    }
}
